OpenSim/Common/STOFileAdapter.h

```diff
@@ -27,28 +27,141 @@
 
 namespace OpenSim {
 
+class STODataTypeNotSupported : public Exception {
+public:
+    STODataTypeNotSupported(const std::string& file,
+                            size_t line,
+                            const std::string& func,
+                            const std::string& datatype) :
+        Exception(file, line, func) {
+        std::string msg = "Datatype '" + datatype + "' is not supported.";
+
+        addMessage(msg);
+    }
+};
+
+class STODataTypeNotFound : public Exception {
+public:
+    STODataTypeNotFound(const std::string& file,
+                        size_t line,
+                        const std::string& func) :
+        Exception(file, line, func) {
+        std::string msg = "DataType not specified in the header.";
+
+        addMessage(msg);
+    }
+};
+
 /** STOFileAdapter is a DelimFileAdapter that presets the delimiters 
-appropriately for STO files.                                                  */
-class OSIMCOMMON_API STOFileAdapter : public DelimFileAdapter {
+appropriately for STO files. The format of the file is as follows:
+\code
+.................................
+.................................
+........ <metadata> .............
```

Comment:
The example does not state that that the metadata is composed of key value pairs. One could imply that but it just seems more clear to state it upfront.